discovery/kubernetes_api_discovery.go

```diff
@@ -222,13 +222,18 @@ func (k *K8sAPIDiscoverer) getServices() ([]byte, error) {
 		return data, err
 	}
 
-	k.lock.Lock()
+	discoveredSvcs := make(map[string]*K8sService)
```

Comment:
Replace the whole `k.discoveredSvcs` each time through. This prevents captured pointers from holding on to large blocks of memory from K8s